beacon-chain/core/helpers/rewards_penalties.go

```diff
@@ -54,9 +56,12 @@ func TotalActiveBalance(state *pb.BeaconState) (uint64, error) {
 //    Increase the validator balance at index ``index`` by ``delta``.
 //    """
 //    state.balances[index] += delta
-func IncreaseBalance(state *pb.BeaconState, idx uint64, delta uint64) *pb.BeaconState {
-	state.Balances[idx] += delta
-	return state
+func IncreaseBalance(state *stateTrie.BeaconState, idx uint64, delta uint64) error {
```

Comment:
Seems expensive and done too many times, instead we can just pass in the balances into this func and return the new, modified balances

Reply:
Then, we can update all balances at the end of the state transition